internal/provider/azure/internal/azureauth/serviceprincipal.go

```diff
@@ -54,18 +45,24 @@ const (
 	AzureUSGovernment = "AzureUSGovernment"
 )
 
-// cloudVaultApps holds the IDs of the Azure Key Vault Application
-// for each cloud type.
-var cloudVaultApps = map[string]string{
-	AzureCloud:        defaultAzureKeyVaultApplicationId,
-	AzureUSGovernment: "7e7c393b-45d0-48b1-a35e-2905ddf8183c",
+// JujuActions contains the allowed actions needed by
+// a Juju controller.
+var JujuActions = []string{
+	"Microsoft.Compute/*",
+	"Microsoft.KeyVault/*",
+	"Microsoft.Network/*",
+	"Microsoft.Resources/*",
+	"Microsoft.Storage/*",
+	"Microsoft.ManagedIdentity/userAssignedIdentities/*",
 }
 
 // MaybeJujuApplicationObjectID returns the Juju Application Object ID
 // if the passed in application ID is the Juju Enterprise App.
+// This is only needed for very old credentials. At some point we
```

Comment:
Can this be left out of 4.0 too?

Reply:
That's what the link was referring to. We need to be 100% sure there's no old credentials still in use first.

Reply:
So best to be conservative for the forward merge - leave it in for now.